pkg/composableschemadsl/compiler/importer.go

```diff
@@ -0,0 +1,54 @@
+package compiler
+
+import (
+	"fmt"
+	"os"
+	"path"
+	"path/filepath"
+
+	"github.com/rs/zerolog/log"
+
+	"github.com/authzed/spicedb/pkg/composableschemadsl/input"
+	"github.com/authzed/spicedb/pkg/genutil/mapz"
+)
+
+type ImportContext struct {
+	pathSegments []string
+	sourceFolder string
+	names        *mapz.Set[string]
+}
+
+const SchemaFileSuffix = ".zed"
+
+func importFile(importContext *ImportContext) (*CompiledSchema, error) {
```

Comment:
why does this take in a `*` vs just the importContext struct itself?

Reply:
Mostly me not having built an intuition yet for when you would use one vs the other.